debezium-server-pubsub/src/main/java/io/debezium/server/pubsub/PubSubChangeConsumer.java

```diff
@@ -213,12 +214,33 @@ void close() {
                 publisher.shutdown();
             }
             catch (Exception e) {
-                LOGGER.warn("Exception while closing publisher: {}", e);
+                LOGGER.warn("Exception while closing publisher: {}", e.getMessage(), e);
             }
         });
+        shutdownChannel(channel);
+    }
 
+    void shutdownChannel(ManagedChannel channel) {
         if (channel != null && !channel.isShutdown()) {
-            channel.shutdown();
+            ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
Please use `io.debezium.util.Threads` class

Reply:
The io.debezium.util.Threads class is designed to work with SourceConnector, but I want to utilize it within a ChangeConsumer context.

Reply:
@ankurg03 Right, could you please then try to create the executor in the way that the name of the executor thread is more desciptive in the sense that it is clear that it is

* from Debezium
* to which sink it relates
* what is its purpose

Maybe you can even create a limited equivalent of `Threads` class in Debezim Server core module so we can re-use it elswhere if possible.

Reply:
Is there a reason for ThreadFactory to extend SourceConnector in the Threads class? The current syntax is as follows:

ThreadFactory threadFactory(Class<? extends SourceConnector> connector, String connectorId, ....)
Would it be better to change the syntax to something like this:

public static ThreadFactory threadFactory(Class<?> connectorType, String connectorId, ....)
This way, the same class can be used for both sink and source processors.

Reply:
It was intentinal as we expected that `SourceConnector` class might provide some common metadata in future we'd rely on. In the hindsight it is probably unnecessary so yet, it could be converted to generic class. In that case I'd also rename argument from `connector` to `component` to reflect its more generic state.

Reply:
Sure, I will raise a pull request to implement this change. i.e. I will make the ThreadFactory in the Threads class more generic.

Reply:
I've updated the PR. Could you please take a look?